packages/api/src/routes/guilds/[guildId]/cases/post.ts

```diff
@@ -0,0 +1,64 @@
+import Joi = require('@hapi/joi');
+import { Request, Response, NextHandler } from 'polka';
+import { injectable } from 'tsyringe';
+import Route from '../../../../Route';
+import { authorize, validate } from '../../../../middleware';
+import CaseManager, { Case } from '../../../../managers/CaseManager';
+import bodyParser from '../../../../middleware/bodyParser';
+
+interface CasesPostBody {
+	cases: Case[];
+}
+
+@injectable()
+export default class CreateCaseRoute implements Route {
+	public middleware = [
+		bodyParser,
+		validate(Joi.object({
+			cases: Joi.array().items(Joi.object({
+				action: Joi.number().positive().integer().max(6),
+				roleId: Joi.when('type', {
+					is: 'role',
+					then: Joi.string().required().pattern(/[0-9]+/),
+					otherwise: Joi.forbidden(),
+				}),
+				actionExpiration: Joi.when('type', {
+					is: Joi.valid('role', 'ban'),
+					then: Joi.date(),
+					otherwise: Joi.forbidden(),
+				}),
+				reason: Joi.string().required(),
+				targetId: Joi.string().pattern(/[0-9]+/).required(),
+				deleteMessageDays: Joi.when('type', {
+					is: Joi.valid('ban', 'softban'),
+					then: Joi.number().positive().max(7).default(1),
+					otherwise: Joi.forbidden(),
+				}),
+				contextMessageId: Joi.string().pattern(/[0-9]+/),
+				referenceId: Joi.number(),
+			})),
+		}).required()),
+		authorize,
+	];
+
+	constructor(
+		public readonly caseManager: CaseManager,
+	) {}
+
+	public async handle(req: Request, res: Response, next: NextHandler) {
+		if (!req.body || !req.userId) return next(new Error('uh oh, something broke'));
```

Comment:
This shouldn't happen and, if it does, I wouldn't know what the error would be. It would probably be a fundamental internal server error. I put this in for type checking and to ensure that, if it does happen, the request is terminated. Happy to change it if you have a better idea.

Reply:
Ah, yeah no thats fine then.